src/JustSaying/AwsTools/MessageHandling/Dispatch/HandlerMap.cs

```diff
@@ -0,0 +1,49 @@
+using System;
```

Comment:
Are we still able to subscribe the message message into multiple queues? I know this is something that was once in use, for example you might want different queues and handlers in an app that sent email/sms/push for a certain event.

Reply:
This is a good question, and might require @martincostello's input. There's a few concerns here, subscriptions to SQS/SNS, and actual `IHandlerAsync` registrations.

### Subscriptions ###
The new builder API subscribes message types to topic/queue names by convention, which is what JustSaying uses to register subscriptions/topics etc. From this perspective it's possible to subscribe to the same message to multiple queues using a custom naming convention.

### IHandlerAsync ###

From a handler perspective, there are two new API's:
`AddJustSayingHandler<TMessage,THandler>` and
`AddJustSayingHandlers(IEnumerable<IHandlerAsync<TMessage, THandler>>`

The first does as you'd expect, and registers a single handler with a transient lifetime.
The second wraps up the multiple handlers into a `ListHandler` which is basically a `Task.WhenAll` around the tasks returned by the handlers.

The important bit here is that the ListHandler is itself an `IHandlerAsync`, so it's still registered into DI as a single handler, it just contains multiple handlers. At resolution time, if there is more than one handler registered to a particular message type in the DI container it will throw.

This means that it's not possible to register a multiple handlers for the same message type, and have them execute in different queue scopes. I suspect this might be a missing feature. In the existing world, I think this is possible as `HandlerMap` is scoped per queue instead of globally?

### Possible Solution ###
MassTransit has a concept called [`receive endpoints`](https://masstransit-project.com/usage/consumers.html#consumer) which is a named group of message types that are sent to the same queue. As registration for message type handling and handlers happens on a per-endpoint basis, it's possible to specify specific handlers for specific endpoints.

Do we need something similar? Or do we just need to be able to somehow explicitly state which handler should handle a particular queue's message? Not sure how this would work by convention.

Reply:
I guess this is a wider question than just this PR? Maybe split it out into an issue and we can discuss whether or not we want to change the multiple-handler handling for v7 or not (I suspect not to limit scope).

Reply:
Seems reasonable. Just to be clear though, I think that this PR is removing support for handling multiple queues with the same message type. I know that at some point this feature was being used, which is why I raise it.

Reply:
Ah, I didn't get that nuance. I thought from George's reply the existing wrapper-ing meant it was fine.

Reply:
I think we should be able to support this. We originally changed it because the HandlerMap had a lot of details that leaked into the SqsNotificationListener, so making it stateless seemed like an obvious move.

We have some further change coming along related to grouping queues to allows control over prefetch/multiplexer-size/queue size on a per-(queue/group) basis instead of all or nothing, so let's re-open this when that lands.

Reply:
This is now supported, thanks to @maisiesadler